toolchain/lowering/lowering_handle.cpp

```diff
@@ -79,14 +79,38 @@ auto LoweringHandleArrayValue(LoweringFunctionContext& context,
 auto LoweringHandleAssign(LoweringFunctionContext& context,
                           SemIR::NodeId /*node_id*/, SemIR::Node node) -> void {
   auto [storage_id, value_id] = node.GetAsAssign();
+  auto storage_type_id = context.semantics_ir().GetNode(storage_id).type_id();
+
   // We can assign from either a value expression or a by-copy initializing
   // expression. In either case, the value of the source is the value
   // representation of the target.
   // TODO: Should we use different semantic nodes for those operations?
-  // TODO: Handle the case where the value representation is not the object
-  // representation.
-  context.builder().CreateStore(context.GetLocal(value_id),
-                                context.GetLocal(storage_id));
+  switch (auto rep = SemIR::GetValueRepresentation(context.semantics_ir(),
+                                                   storage_type_id);
+          rep.kind) {
+    case SemIR::ValueRepresentation::None:
+      break;
```

Comment:
I *think* we might want to generate the assign node in this case, but equally, maybe we don't need to? It seems like it might be unnecessary, and then it wouldn't reach this point -- a minor efficiency. And semantically, no assignment actually is expected, right?

Reply:
Okay, makes sense to me.